web/src/components/users/utils.test.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import { suggestUsernames } from "./utils";
+
+describe('suggestUsernames', () => {
+  test('handles basic single name', () => {
+    expect(suggestUsernames('John')).toEqual(expect.arrayContaining(['john', 'j']));
```

Comment:
I think `j` is a kind of useless suggestion. That's why I filtered out the suggestions with less than 3 characters in my above proposal.